### ISSUE TITLE
cvmfs py3-v4.1.0 breaks parasitic builds (Trac #2425)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2417]
This is either a bug in cmake or cvmfs.  Not sure.

here what I'm doing:
built combo V00-00-03 first against py3-v4.1.0
into cmvfs/user
/cvmfs/icecube.opensciencegrid.org/users/blaufuss/combo/V00-00-03/build
then tried the parasitic build
issue:  cmake was not properly getting the PYTHON_INCLUDES etc into the parasitic workspace
and builds fail immediately (can find Python.h)

I stepped back from py3-v4.1.0 to py3-v4.0.1
and built:
/cvmfs/icecube.opensciencegrid.org/users/blaufuss/combo/V00-00-03/build2
and now things work fine
they're different versions of cmake (3.16 vs 3.11)
and python but this is a bit puzzling

Migrated from https://code.icecube.wisc.edu/ticket/2425
```json
{
    "status": "closed", 
    "changetime": "2020-05-12T21:11:09", 
    "description": "This is either a bug in cmake or cvmfs.  Not sure.\n\nhere what I'm doing:\nbuilt combo V00-00-03 first against py3-v4.1.0\ninto cmvfs/user\n/cvmfs/icecube.opensciencegrid.org/users/blaufuss/combo/V00-00-03/build\nthen tried the parasitic build\nissue:  cmake was not properly getting the PYTHON_INCLUDES etc into the parasitic workspace\nand builds fail immediately (can find Python.h)\n\nI stepped back from py3-v4.1.0 to py3-v4.0.1\nand built:\n/cvmfs/icecube.opensciencegrid.org/users/blaufuss/combo/V00-00-03/build2\nand now things work fine\nthey're different versions of cmake (3.16 vs 3.11)\nand python but this is a bit puzzling", 
    "reporter": "blaufuss", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1589317869741189", 
    "component": "combo core", 
    "summary": "cvmfs py3-v4.1.0 breaks parasitic builds", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2020-04-22T19:26:01", 
    "milestone": "", 
    "owner": "nwhitehorn", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20200511T21:02:14: nwhitehorn commented:

> The issue is that newer cmake spells PYTHON_LIBRARY Python_LIBRARY etc. We provide some compat. shims in tools/python.cmake, but those compat shims are in-memory only and are not written to the cmake cache, which the parasitic build stuff reads directly.
> 
> This seems straight-forward, but tedious, to fix.

**Comment by icecube-trac** [comment]
Trac update at 20200511T21:42:09: 
* nwhitehorn changed resolution from "" to "fixed"
* nwhitehorn changed status from "new" to "closed"
* nwhitehorn commented:

> In changeset 180064:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="180064"
> Make sure all Python-related variables are stored in CMakeCache.txt.
> 
> These variables are needed, from the cache, by the parasitic project CMake
> code and, in newer cmake, were stored with different names. This makes sure
> they are stored and that important variables are available with both old-
> and new-style names in both regular and parasitic metaprojects.
> 
> This fixes #2417.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20200512T21:11:09: blaufuss commented:

> Confirmed. fixed with combo/trunk builds and combo/cmake/trunk in my parasitic project.
> 
> Thanks Nathan
